src/ucs/config/parser.c

```diff
@@ -1021,20 +1035,36 @@ static ucs_status_t ucs_config_apply_env_vars(void *opts, ucs_config_field_t *fi
             strncpy(buf + prefix_len, field->name, sizeof(buf) - prefix_len - 1);
             env_value = getenv(buf);
             if (env_value != NULL) {
-                ucs_config_parser_release_field(field, var);
-                ucs_config_parser_mark_env_var_used(buf);
-                status = ucs_config_parser_parse_field(field, env_value, var);
-                if (status != UCS_OK) {
-                    /* If set to ignore errors, restore the default value */
-                    ucs_status_t tmp_status =
-                        ucs_config_parser_parse_field(field, field->dfl_value,
-                                                      var);
-                    if (ignore_errors) {
-                        status = tmp_status;
+                status = ucs_config_parser_mark_env_var_used(buf);
+
+                if (ucs_config_is_deprecated_field(field)) {
+                    if ((status == UCS_OK) &&
+                        ucs_global_opts.warn_unused_env_vars) {
+                        ucs_warn("deprecated env variable %s used "
+                                 "(set %s%s=n to suppress this warning)",
+                                 buf, UCS_CONFIG_PREFIX,
+                                 UCS_GLOBAL_OPTS_WARN_UNUSED_CONFIG);
+                    }
+                } else {
+                    if ((status != UCS_OK) &&
```

Comment:
why do we even need to check the status here? can just ignore it like before..

Reply:
I think that Coverity will be not happy with that, because we will set the new value `status` variable w/o touching/using it before
```
status = ucs_config_parser_mark_env_var_used(buf);
if ((status != UCS_OK) &&
    (status != UCS_ERR_ALREADY_EXISTS)) {
    ucs_warn(...);
    continue;
}

... // the same code
```

Reply:
discussed offline with @yosefe:
will refactor `ucs_config_parser_mark_env_var_used` to return the value from the argument and check inside `if` and don't check inside `else`